aws-java-sdk-core/src/main/java/com/amazonaws/auth/policy/internal/JsonPolicyReader.java

```diff
@@ -146,9 +146,17 @@ private Statement statementOf(JsonNode jStatement) {
         if (isNotNull(actionNodes))
             statement.setActions(actionsOf(actionNodes));
 
+        List<Resource> resources = new LinkedList<Resource>();
         JsonNode resourceNodes = jStatement.get(JsonDocumentFields.RESOURCE);
         if (isNotNull(resourceNodes))
-            statement.setResources(resourcesOf(resourceNodes));
+            resources.addAll(resourcesOf(resourceNodes, false));
+
+        JsonNode notResourceNodes = jStatement.get(JsonDocumentFields.NOT_RESOURCE);
+        if (isNotNull(notResourceNodes))
+            resources.addAll(resourcesOf(notResourceNodes, true));
```

Comment:
Is it valid to have both `RESOURCE` and `NOT_RESOURCE` type in same policy? If so, can you add more tests having both types?